### ISSUE TITLE
Update Grafana Version 7

### ISSUE THREAD

**Issue by ricetj** [department-of-veterans-affairs/va.gov-team#12322]
## Description
The Identity team and BE tools team would like to utilize some features in Grafana 7 (latest). This will be a major upgrade from 6 to 7 and it looks like there are going to be some issues with w/ backend plugins & signing. 

## Background/context/resources
Both Bennie and Keifer are willing to support this rollout.  

- Ref: https://dsva.slack.com/archives/CJYRZK2HH/p1597078047032300
- Ref: https://github.com/department-of-veterans-affairs/devops/pull/7386#event-3639744604

## Technical notes

 - Upgrade guide: https://grafana.com/docs/grafana/latest/installation/upgrading/#upgrading-to-v70
-  ref: https://dsva.slack.com/archives/CJYRZK2HH/p1597089144037000?thread_ts=1597078047.032300&cid=CJYRZK2HH
---
## Tasks
- [x] Test out the version upgrade 
- [x] Upgrade the version of Grafana

## Definition of Done
- [x] Grafana is using version 7 and any issues are addressed that come up from the upgrade.
- [x] Document upgrade process for future onboarding

---
### Reminders
- [x] Please attach your team label and any other appropriate label(s) 
- [ ] Please attach the needs grooming tag if needed
- [ ] Please connect to an epic

**Comment by pnwstevan** [comment]
### **Detailed documentation of upgrade procedure for future reference/onboarding**

**tl;dr**
Detailed documentation of first production deploy; process itself and documentation is likely excessive, but consider this essentially `upgrading Grafana the hard way` as a bit of a forcing function to learn a bunch of things.  Hopefully this is helpful for future reference and onboarding other Ops team members.

**Thoughts/Notes:**
- Seems like some opportunity to optimize the changes to seed job schedule; a bit of manual work to reconcile the intended/actual state.
- Felt a bit blind when pulling the trigger, other than waiting for Grafana to BRD and manually checking things out; likely some opportunity for improved visibility with logging.
- Should have potentially suppressed any alerts?
- Consider making use of deployment/annotation API in Grafana for historical record of upgrade (ie: change) in timeseries graphs.
 
**Prerequisites**:

Access and Auth:
- [ ] SSH keys and AWS Access as noted [here](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/working-with-vsp/orientation/request-access-to-tools.md#2-request-that-your-ssh-keys-be-authorized-so-that-you-can-use-the-developer-tools-such-as-jenkins-grafana-and-sentry)
- [ ] GitHub Access
  - [ ] Added to the VA Org
  - [ ] Commit bit on the [devops repo](https://github.com/department-of-veterans-affairs/devops)
- [ ] AWS Setup
  - [ ] [Keys and MFA](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/backend/aws-account-setup.md)
  - [ ] [DVSA AWS Console](https://dsvagovcloud.signin.amazonaws-us-gov.com/console) Access
- [ ] Everything in the [internal tools doc](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/internal-tools.md) setup and working
  - [ ] Socks Proxy Access
  - [ ] Applications
    - [ ] [Grafana](http://grafana.vfs.va.gov/) 
    - [ ] [Jenkins](http://jenkins.vfs.va.gov/) 
	 
Local Tools for Dev & Testing:
- [ ] Git
- [ ] Vagrant
- [ ] VirtualBox
- [ ] Docker
- [ ] Ansible
- [ ] Python Stuffs (Pyenv, pip, Python 3.x, 2.x, etc..)
 - [ ] AWS CLI
 - [ ] aws-mfa tool (enables 2MFA for access to AWS resources via CLI)
 - [ ] credstash


**Upgrade Steps:**
1. Ensure all preqs and access per above is setup - better to do this now than get blocked in the middle of a workflow.

2. Pre-work - gather context, requirements, etc..
  -  Reviewed the [initial request](https://github.com/department-of-veterans-affairs/devops/pull/7386).
  - Let the team that owns the app know [I'll be working on it](https://dsva.slack.com/archives/CJYRZK2HH/p1597867272193600).
  - Reviewed the Grafana [Releases](https://github.com/grafana/grafana/releases) and [Release Notes](https://community.grafana.com/t/release-notes-v7-1-x/32967).
  - Note: I determined there had been a version bump since the request which looked like it made sense to upgrade to.

3. Making the changes for upgrade
   - Clone the master branch of the respective repo locally (in this case, Grafana configs are in the [devops repo](https://github.com/department-of-veterans-affairs/devops))
   - Create and checkout a new local branch
   - Make the respective changes to enable upgrade:
	    - Version bump in Ansible [config](ansible/build/roles/grafana-vagov/vars/main.yml) for Grafana


4. Test the upgrade locally
	  - Ensure you have a valid AWS token and working CLI
	    - Can test with `aws sts get-caller-identity` 
      - Review and follow [guide for using Vagrant](https://github.com/department-of-veterans-affairs/devops/tree/master/ansible/vagrant)
	  - Go to your local working directory with checked out branch then `~/devops/ansible`
	  - Setup Python 
	    - Virtual Environment
		- `pip install -r requirements.txt`
	   - Build with changes: `export APP=grafana-vagov;export ENV=vagov-dev; export REF=master; vagrant up --provision-with build`
	   - Forward port to connect locally: `vagrant ssh -- -L 3000:localhost:3000`
	   - Login page @ http://localhost:3000 - yay!  
	     - Boo - can't login cause no DB :(
	   - Reset admin password!
	   - From SSH shell into Vagrant Host:
             - Find container: `docker ps`
               - Shell into container: `docker exec -ti <container id> bash`
	       - Reset admin password: `grafana-cli admin reset-admin-password <new password`
	   - Can log in to locally upgraded test build!
	   - Notice there are warnings about unsigned plugins, which I saw in release notes and need to make another change.
	   - Destroy Vagrant build: `vagrant destroy`
	   - Make [change](https://github.com/department-of-veterans-affairs/devops/pull/7386/commits/20a70da205ec3631acafd3b14f11e4985a90deaf) to allow unsigned plugins
	   - Rebuild again with above steps
	   - SUCCESS! :D
	   - Don't submit a PR yet, since it will get built, deployed and released automatically per Jenkins schedule (more on this below).


5. Prepare for Production Upgrade
	- Since there is **some** risk, and my first deploy going to be extra cautious and do this manually.
	- Disable daily build for Grafana by removing cron in seed job:
	  - [Commit](https://github.com/department-of-veterans-affairs/devops/pull/7457/files)
	  - Push commit to master
	  - Submit [Pull Request](https://github.com/department-of-veterans-affairs/devops/pull/7457/)
    - Also need to make this change on live production server:
      - Login to J[http://jenkins.vfs.va.gov/](http://jenkins.vfs.va.gov/)
      - Seed Job->Configure-DSL Script
        - Make same change as above PR
   - NOW we can submit our actual upgrade PR, without fear it will be deployed automatically.
     - [Commit](https://github.com/department-of-veterans-affairs/devops/pull/7386/commits/20a70da205ec3631acafd3b14f11e4985a90deaf) and push to master
       - Submit [Pull Request](https://github.com/department-of-veterans-affairs/devops/pull/7386) for Upgrade
   - Discuss with team in office hours; plan for upgrade 10am PST 8/25
   - Announce ^ via Slack to original [request thread](https://dsva.slack.com/archives/CJYRZK2HH/p1597867272193600), and in [#vfs-engineers](https://dsva.slack.com/archives/C0MQ281DJ/p1598301421011100)


6. Production Upgrade
  - Get browser tabs open/ready:
    - Jenkins Prod
    - Grafana
    - AWS Console:
      - RDS (for Prod DB Backup / Monitoring)
      - CloudWatch (created a simple dashboard for network/io from ELB serving Grafana)
- Build, Release & Deploy time!
  - Backup Grafana Prod DB
    - Find prod grafana DB in RDS
    - Actions->Take Snapshot
    - Confirm snapshot is complete
    - Leave tab open for DB metrics in case needed for troubleshooting
  - Build, Release, Deploy Grafana upgrade:
    - Login to Jenkins
    - BUILD!
      - Build->grafana-vagov
      - Build with Parameters
      - Click on active Build #
      - Monitor build with "Console Output"
      - When complete, click link in "Scheduling Project:"
    - RELEASE!
      - Click on active Release #
      - Monitor release with "Console Output"
    - DEPLOY!
        - Click on active Deploy #
        - Monitor deploy with "Console Output"
	- If successful, app should be up, which it was.

7. Testing / Validation
  - Logged into Grafana
  - Checked a bunch of charts
  - Looked for anything obviously broken
  - Monitored CW metrics and RDS charts
 
8. Post Deploy / Clean-up
  - Notification upgrade was complete in previous Slack threads
  - Re-enable seed job:
  - Revert [commit](https://github.com/department-of-veterans-affairs/devops/pull/7386) (could have submitted new commit/PR, thought this was cleaner)
  - Re-enabled it on live Jenkins server (see above)

**Comment by pnwstevan** [comment]
@ricetj Docs as discussed (CC: @dginther).

Is there a SOP for closing issues - do you/PMs typically do that?  When they're done, or at end of sprint?

**Comment by ricetj** [comment]
> @ricetj Docs as discussed (CC: @dginther).
> 
> Is there a SOP for closing issues - do you/PMs typically do that? When they're done, or at end of sprint?

Hey Stevan, we have been having folks close their own tickets after completing all the AC. In terms of an SOP we do not have that.
